ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py

```diff
@@ -339,3 +339,42 @@ def should_connect_to_wan_rs(self: SharedUtils, path_groups: list) -> bool:
             local_path_group["name"] in path_groups and any(wan_interface["connected_to_pathfinder"] for wan_interface in local_path_group["interfaces"])
             for local_path_group in self.wan_local_path_groups
         )
+
+    @cached_property
+    def wan_ha(self: SharedUtils) -> bool:
+        """
+        Only trigger HA if 2 devices are in the same group and ha.enabled is true
+        """
+        if self.cv_pathfinder_role in [None, "pathfinder"]:
+            return False
+        return get(self.wan_site, "ha.enabled", default=True) and len(self.switch_data_node_group_nodes) == 2
+
+    @cached_property
+    def wan_ha_path_group_name(self) -> str:
+        """
+        Return HA path group name for the WAN design.
+        Used in both network services and overlay python modules.
+
+        TODO make this configurable
+        """
+        return "LAN_HA"
+
+    @cached_property
+    def wan_ha_interfaces(self) -> list:
+        """
+        Return list of interfaces for HA
+
+        Used in underlay and overlay or maybe yes
+        """
+        return [uplink for uplink in self.get_switch_fact("uplinks") if get(uplink, "vrf") is None]
```

Comment:
HA does not work if the interfaces have ip address DHCP (since you need to configure the tunnel statically). Should there be a check somewhere that skips LAN interfaces if they're configured via DHCP? (for cloudEOS, we could have such interfaces, although I suppose we wouldn't typically enable HA on such sites).

Reply:
AVD does not support DHCP on "uplink_interfaces" which is what we use to designate the LAN interfaces on the WAN routers.